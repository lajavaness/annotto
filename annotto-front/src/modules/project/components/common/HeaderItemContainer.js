import { useTranslation } from 'react-i18next'
import PropTypes from 'prop-types'
import React from 'react'

import * as Styled from './__styles__/HeaderItemContainer.styles'

const HeaderItemContainer = ({ id }) => {
	const { t } = useTranslation('project')

	return (
		<Styled.Root>
			<Styled.Label>{t('project:annotation.itemName', { id })}</Styled.Label>
		</Styled.Root>
	)
}

export default HeaderItemContainer

HeaderItemContainer.propTypes = {
	/** Defines the id of the item. */
	id: PropTypes.string,
}

HeaderItemContainer.defaultProps = {
	id: null,
}
