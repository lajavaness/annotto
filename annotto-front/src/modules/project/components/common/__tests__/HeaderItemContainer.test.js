import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import React from 'react'

import theme from '__theme__'

import HeaderItemContainer from '../HeaderItemContainer'

const defaultProps = {}
const getInstance = (props = {}) => (
	<ThemeProvider theme={theme}>
		<HeaderItemContainer {...defaultProps} {...props} />
	</ThemeProvider>
)

describe('HeaderItemContainer', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(getInstance())
		expect(asFragment()).toMatchSnapshot()
	})
})
