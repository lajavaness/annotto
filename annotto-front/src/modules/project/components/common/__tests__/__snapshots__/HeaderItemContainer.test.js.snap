// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderItemContainer matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gsBqHK cBlUHF"
  >
    <span
      class="sc-aKXVD"
    >
      project:annotation.itemName
    </span>
  </div>
</DocumentFragment>
`;
